src/core/checkpoints/index.ts

```diff
@@ -285,21 +291,48 @@ export async function checkpointDiff(task: Task, { ts, previousCommitHash, commi
 
 	TelemetryService.instance.captureCheckpointDiffed(task.taskId)
 
-	let prevHash = commitHash
-	let nextHash: string | undefined = undefined
+	let fromHash: string | undefined
+	let toHash: string | undefined
+	let title: string
 
-	if (mode !== "full") {
-		const checkpoints = task.clineMessages.filter(({ say }) => say === "checkpoint_saved").map(({ text }) => text!)
-		const idx = checkpoints.indexOf(commitHash)
-		if (idx !== -1 && idx < checkpoints.length - 1) {
-			nextHash = checkpoints[idx + 1]
-		} else {
-			nextHash = undefined
-		}
+	const checkpoints = task.clineMessages.filter(({ say }) => say === "checkpoint_saved").map(({ text }) => text!)
+
+	if (["from-init", "full"].includes(mode) && checkpoints.length < 1) {
+		vscode.window.showInformationMessage("No first checkpoint to compare.")
+		return
+	}
+
+	const idx = checkpoints.indexOf(commitHash)
+	switch (mode) {
+		case "checkpoint":
+			fromHash = commitHash
+			toHash = idx !== -1 && idx < checkpoints.length - 1 ? checkpoints[idx + 1] : undefined
+			title = "Changes compare with next checkpoint"
```

Comment:
These titles should be translated, and I think this one should say “compared with next checkpoint” in English